patches/minecraft/net/minecraft/client/particle/ParticleDigging.java.patch

```diff
@@ -0,0 +1,30 @@
+--- ../src-base/minecraft/net/minecraft/client/particle/ParticleDigging.java
++++ ../src-work/minecraft/net/minecraft/client/particle/ParticleDigging.java
+@@ -33,6 +33,7 @@
+     public ParticleDigging func_174846_a(BlockPos p_174846_1_)
+     {
+         this.field_181019_az = p_174846_1_;
++        updateTexture();
+ 
+         if (this.field_174847_a.func_177230_c() == Blocks.field_150349_c)
+         {
+@@ -49,6 +50,7 @@
+     {
+         this.field_181019_az = new BlockPos(this.field_187126_f, this.field_187127_g, this.field_187128_h);
+         Block block = this.field_174847_a.func_177230_c();
++        updateTexture();
+ 
+         if (block == Blocks.field_150349_c)
+         {
+@@ -123,4 +125,11 @@
+                 return (new ParticleDigging(p_178902_2_, p_178902_3_, p_178902_5_, p_178902_7_, p_178902_9_, p_178902_11_, p_178902_13_, Block.func_176220_d(p_178902_15_[0]))).func_174845_l();
+             }
+         }
++
++    private void updateTexture()
++    {
++        IBlockState state = this.field_174847_a.func_185899_b(this.field_187122_b, this.field_181019_az);
++        state = state.func_177230_c().getExtendedState(state, this.field_187122_b, this.field_181019_az);
++        this.func_187117_a(Minecraft.func_71410_x().func_175602_ab().func_175023_a().func_178122_a(state));
```

Comment:
You need to add sanity to this because the position WILL NOT BE the position of the block that is being broken.
So you need to make sure that state == sourceState before you even attempt to extend it.